packages/nbdime/src/common/mergeview.ts

```diff
@@ -143,563 +341,404 @@ function createNbdimeMergeView(
-      }
-    }
-    this.baseEditor.on('change', change);
-    this.ownEditor.on('change', change);
-    this.baseEditor.on('markerAdded', setDealign);
-    this.baseEditor.on('markerCleared', setDealign);
-    this.ownEditor.on('markerAdded', setDealign);
-    this.ownEditor.on('markerCleared', setDealign);
-    this.baseEditor.on('viewportChange', function() { set(false); });
-    this.ownEditor.on('viewportChange', function() { set(false); });
-    update();
-    return update;
-  }
-
   protected modelInvalid(): boolean {
     return this.model instanceof DecisionStringDiffModel &&
             this.model.invalid;
   }
 
-  protected onGutterClick(instance: CodeMirror.Editor, line: number, gutter: string, clickEvent: Event): void {
-    if ((clickEvent as MouseEvent).button !== 0) {
-      // Only care about left clicks
-      return;
-    }
-    let li = instance.lineInfo(line);
-    if (!li.gutterMarkers || !li.gutterMarkers.hasOwnProperty(gutter)) {
-      return;
-    }
-    let node = li.gutterMarkers[gutter];
-    if (node && node.sources) {
-      let ss = node.sources as ChunkSource[];
-      if (gutter === GUTTER_PICKER_CLASS) {
-        if (instance === this.ownEditor) {
-          for (let s of ss) {
-            s.decision.action = s.action;
-          }
-        } else if (this.type === 'merge' && instance === this.baseEditor) {
-          for (let s of ss) {
-            s.decision.action = 'base';
-          }
-        }
-        for (let i=ss.length - 1; i >= 0; --i) {
-          let s = ss[i];
-          if (this.type === 'merge' && hasEntries(s.decision.customDiff)) {
-            // Custom diffs are cleared on pick,
-            // as there is no way to re-pick them
-            s.decision.customDiff = [];
-          }
-        }
-        if (ss.length === 0) {
-          // All decisions empty, remove picker
-          // In these cases, there should only be one picker, on base
-          // so simply remove the one we have here
-          instance.setGutterMarker(line, GUTTER_PICKER_CLASS, null);
-        }
-      } else if (gutter === GUTTER_CONFLICT_CLASS) {
-        for (let s of ss) {
-          s.decision.conflict = false;
-        }
-      }
-      for (let dv of this.baseEditor.state.diffViews as DiffView[]) {
-        if (dv.model instanceof DecisionStringDiffModel) {
-          dv.model.invalidate();
-        }
-        dv.forceUpdate('full');
-      }
-    }
-  }
-
-  protected registerScroll(): void {
-    let self = this;
-    this.baseEditor.on('scroll', function() {
-      self.syncScroll(EventDirection.OUTGOING);
-    });
-    this.ownEditor.on('scroll', function() {
-      self.syncScroll(EventDirection.INCOMING);
-    });
-  }
-
-  /**
+ /**
    * Sync scrolling between base and own editors. `type` is used to indicate
    * which editor is the source, and which editor is the destination of the sync.
    */
-  protected syncScroll(type: EventDirection): void {
```

Comment:
- The comment should be updated to reflect the changes
- The method should be private

Reply:
Thanks. This has been modified.